src/workspaces/SettingsModal/BucketLifecycleSettings.tsx

```diff
@@ -94,7 +82,7 @@ const BucketLifecycleSettings = (props: BucketLifecycleSettingsProps): ReactNode
             }
           }}
           options={prefixOptions()}
-          isDisabled={!lifecycleRulesEnabled}
+          isDisabled={!lifecycleRulesEnabled || !isOwner}
```

Comment:
This is a minor bug fix from the last [PR](https://github.com/DataBiosphere/terra-ui/pull/5036#discussion_r1744101869) (test coverage has been added for it).